### PR TITLE
icuregex: Update to ICU 73

### LINKED ISSUE #13911: Feature Request: Upgrade to ICU 73
### Feature Description

MySQL 8.0.34 and 8.1.0 have updated to bundling ICU 73 instead of ICU 69. We should follow suit.

### Use Case(s)

Ensure to match with latest ICU that MySQL uses for regular expression support.

### PR DESCRIPTION
**Pull request by dbussink**
This updates the ICU data to ICU 73. The most significant change here is that in ICU 70 the emoji properties were split off into a separate file, so we need to include that file here too and add a loader for it.

Additionally the regex code was reviewed from 69 to 73 to see if there's any significant differences. There were a few typo fixes in comments and some crash assertions were changed to errors instead.

MySQL also updated to bundling ICU 73 with the 8.0.34 release.

## Related Issue(s)

Fixes #13911

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/mysql/icuregex/perl_test.go [hunk 1]
```diff
@@ -27,22 +27,25 @@ import (
 	"strconv"
 	"strings"
 	"testing"
+
+	"github.com/stretchr/testify/require"
```
Comment (systay) [inline go/mysql/icuregex/perl_test.go:31]: yay!

File: go/mysql/icuregex/internal/uemoji/loader.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package uemoji
```
Comment (systay) [inline go/mysql/icuregex/internal/uemoji/loader.go:1]: do we want a license on these files?

**Review by systay — APPROVED** [review]
LGTM

File: go/mysql/icuregex/internal/uemoji/loader.go (hunk 2 above)
Comment (dbussink) [reply go/mysql/icuregex/internal/uemoji/loader.go:22]: Good point, fixed!

**Pull request merged**